test/controllers/articles_controller_test.rb

```diff
@@ -4,18 +4,65 @@
 
 # Controller tests for news articles
 class ArticlesControllerTest < FunctionalTestCase
-  def test_create_article_get
+  # Prove anyone can see (complete) article index
+  def test_index
+    get(:index)
+    assert(:success)
+    # Prove index (or redirect) includes link to first article
+    assert_select("a[href *= '#{Article.first.id.to_s}']")
+  end
+
+  def test_index_article
+    # Prove any user can see article index (based on current query)
+    get(:index_article)
+    assert(:success)
+  end
+
+  def test_index_article_ui
+    # Prove privileged users get link to create an article
+    login(users(:article_writer).login)
+    get(:index_article)
+    assert_select("a", { text: :create_article_title.l },
+                  "Page is missing a link to :create")
+  end
```

Comment:
This test fails.
I've changed my mind about how to deal with assertions that fail because the UI is incomplete. Instead of commenting them out and adding a TODO, I'll isolate them from passing assertions by putting them into separate methods.
That will make it easier to find out where the UI needs help (or the assertions need to be improved).